src/main/java/com/example/rezerwowy/controllers/RoleController.java

```diff
@@ -0,0 +1,56 @@
+package com.example.rezerwowy.controllers;
+
+import com.example.rezerwowy.dtos.RoleDto;
+import com.example.rezerwowy.mappers.RoleMapper;
+import com.example.rezerwowy.models.Role;
+import com.example.rezerwowy.services.RoleService;
+import jakarta.validation.Valid;
+import lombok.RequiredArgsConstructor;
+import org.springframework.http.HttpStatus;
+import org.springframework.http.ResponseEntity;
+import org.springframework.web.bind.annotation.*;
+
+import java.util.Collection;
+
+@RestController
+@RequestMapping("roles")
+@RequiredArgsConstructor
+public class RoleController {
+    private final RoleService roleService;
+    private final RoleMapper roleMapper;
+
+    @GetMapping
+    public ResponseEntity<Collection<RoleDto>> getRoles() {
+        Collection<Role> roles = roleService.getRoles();
+        Collection<RoleDto> roleDtos = roles.stream()
+                .map(roleMapper::mapRoleToRoleDto)
+                .toList();
+
+        return ResponseEntity.status(HttpStatus.OK).body(roleDtos);
+    }
+
+    @GetMapping("/{id}")
+    public ResponseEntity<RoleDto> getRoleById(@PathVariable("id") Long id) {
+        Role role = roleService.getRoleById(id);
+        RoleDto roleDto = roleMapper.mapRoleToRoleDto(role);
+
+        return ResponseEntity.status(HttpStatus.OK).body(roleDto);
+    }
+
+    @PostMapping
+    public ResponseEntity<RoleDto> createRole(@RequestBody @Valid RoleDto roleDto) {
+        Role roleToCreate = roleMapper.mapRoleDtoToRole(roleDto);
+        Role createdRole = roleService.createRole(roleToCreate);
+        RoleDto createdRoleDto = roleMapper.mapRoleToRoleDto(createdRole);
+
+        return ResponseEntity.status(HttpStatus.CREATED).body(createdRoleDto);
+    }
+
+    @DeleteMapping("/{id}")
+    public ResponseEntity<Void> deleteRole(@PathVariable("id") Long id) {
```

Comment:
HTTP DELETE Response Codes
A successful response MUST be 200 (OK) if the server response includes a message body, 202 (Accepted) if the DELETE action has not yet been performed, or 204 (No content) if the DELETE action has been completed but the response does not have a message body.

Reply:
tak tylko wrzucam, ale nie czepiam sie jak cos xd